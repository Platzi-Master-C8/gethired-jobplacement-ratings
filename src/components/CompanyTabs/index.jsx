import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Tabs, Tab, Box } from '@mui/material';

const TabPanel = ({ children, value, index }) => {
    return (
        <div
            role="tabpanel"
            hidden={value !== index}
            id={`simple-tabpanel-${index}`}
            aria-labelledby={`simple-tab-${index}`}
        >
            {value === index && <Box sx={{ p: 3 }}>{children}</Box>}
        </div>
    );
};

TabPanel.propTypes = {
    children: PropTypes.node.isRequired,
    index: PropTypes.number.isRequired,
    value: PropTypes.number.isRequired,
};

const CompanyTabs = ({ tabsOptions }) => {
    const [value, setValue] = useState(1);

    const handleChange = (_event, newValue) => {
        setValue(newValue);
    };

    return (
        <Box sx={{ width: '100%' }}>
            <Box>
                <Tabs value={value} onChange={handleChange}>
                    {tabsOptions.map((option) => (
                        <Tab key={`TabButton-${option.tabKey}`} label={option.tabKey} />
                    ))}
                </Tabs>
            </Box>
            {tabsOptions.map((option, i) => (
                <TabPanel key={`TabContent-${option.tabKey}`} value={value} index={i}>
                    {option.tabContent}
                </TabPanel>
            ))}
        </Box>
    );
};

CompanyTabs.propTypes = {
    tabsOptions: PropTypes.arrayOf(
        PropTypes.shape({
            tabKey: PropTypes.string.isRequired,
            tabContent: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
        }),
    ).isRequired,
};

export default CompanyTabs;
