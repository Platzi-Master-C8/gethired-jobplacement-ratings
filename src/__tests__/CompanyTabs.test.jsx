import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';

import CompanyTabs from '../components/CompanyTabs';

const tabsOptions = [
    { tabKey: 'Overview', tabContent: 'Overview' },
    { tabKey: 'Reviews', tabContent: 'Reviews' },
    { tabKey: 'Jobs', tabContent: 'Jobs' },
];

const defaultProps = { tabsOptions };
const renderComponent = (props = {}) => render(<CompanyTabs {...defaultProps} {...props} />);

describe('<CompanyTabs />', () => {
    test('Component render', () => {
        const { asFragment } = renderComponent();
        expect(asFragment()).toMatchSnapshot();
    });

    test('Change tabs', () => {
        const { getByText, getByRole } = renderComponent();

        fireEvent.click(getByText(tabsOptions[0].tabKey));
        expect(getByRole('tab', { name: tabsOptions[0].tabKey, selected: true }));
        expect(getByRole('tab', { name: tabsOptions[1].tabKey, selected: false }));
        expect(getByRole('tab', { name: tabsOptions[2].tabKey, selected: false }));

        fireEvent.click(getByText(tabsOptions[1].tabKey));
        expect(getByRole('tab', { name: tabsOptions[0].tabKey, selected: false }));
        expect(getByRole('tab', { name: tabsOptions[1].tabKey, selected: true }));
        expect(getByRole('tab', { name: tabsOptions[2].tabKey, selected: false }));

        fireEvent.click(getByText(tabsOptions[2].tabKey));
        expect(getByRole('tab', { name: tabsOptions[0].tabKey, selected: false }));
        expect(getByRole('tab', { name: tabsOptions[1].tabKey, selected: false }));
        expect(getByRole('tab', { name: tabsOptions[2].tabKey, selected: true }));
    });
});
