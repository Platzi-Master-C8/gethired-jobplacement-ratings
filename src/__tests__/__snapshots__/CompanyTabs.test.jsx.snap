// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CompanyTabs /> Component render 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-8atqhb"
  >
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiTabs-root css-1ujnqem-MuiTabs-root"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
          style="overflow: hidden; margin-bottom: 0px;"
        >
          <div
            class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Overview
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="0"
              type="button"
            >
              Reviews
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Jobs
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <span
            class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
    </div>
    <div
      aria-labelledby="simple-tab-0"
      hidden=""
      id="simple-tabpanel-0"
      role="tabpanel"
    />
    <div
      aria-labelledby="simple-tab-1"
      id="simple-tabpanel-1"
      role="tabpanel"
    >
      <div
        class="MuiBox-root css-19kzrtu"
      >
        Reviews
      </div>
    </div>
    <div
      aria-labelledby="simple-tab-2"
      hidden=""
      id="simple-tabpanel-2"
      role="tabpanel"
    />
  </div>
</DocumentFragment>
`;
